python_modules/libraries/dagster-aws/dagster_aws/pipes.py

```diff
@@ -241,3 +267,144 @@ def run(
 
         # should probably have a way to return the lambda result payload
         return PipesClientCompletedInvocation(session)
+
+
+class PipesGlueContextInjector(PipesS3ContextInjector):
+    def no_messages_debug_text(self) -> str:
+        return "Attempted to inject context via Glue job arguments."
+
+
+class PipesGlueLogsMessageReader(PipesCloudWatchMessageReader):
+    def no_messages_debug_text(self) -> str:
+        return "Attempted to read messages by extracting them from the tail of CloudWatch logs directly."
```

Comment:
What is the role of this class? Looks redundant with `PipesCloudWatchMessageReader`.

Reply:
That's a good question! 

The reason I've added these classes is because that's how the Lambda client was set up with `PipesLambdaEventContextInjector`. `PipesLambdaEventContextInjector` is also just an alias for `PipesEnvContextInjector`. I've discussed this with @cmpadden and we agreed to replicate the setup for Glue. 

I think this was done to provide a default (and obvious) `MessageReader` for the user. This `MessageReader` should be as native as possible and work out of the box with the given external process. 

This is what `PipesCloudWatchMEssageReader` is supposed to be for Glue, but we decided to implement it in a separate PR. So right now it's basically a mock. 

I will split this PR into more meaningful pieces and remove this mock from the current PR. 

But also let me know if you don't want to keep the practice of providing default MessageReader/ContextInjector classes per Pipes client. In this case, we can deprecate the `PipesLambdaEventContextInjector` and use the common classes instead.